ftplugin/c.vim

```diff
@@ -1,6 +1,8 @@
 let s:cb = {->
-	 \ switchy#has_suffix(bufname(''), '.c')
-         \ ? bufname('')[:-3] .. '.h'
-         \ : bufname('')[:-3] .. '.c'}
+        \ switchy#has_suffix(bufname(''), '.h')
+        \ ? filereadable(bufname('')[:-3] .. '.c' ) ?  bufname('')[:-3] .. '.c' :  bufname('')[:-3] .. '.cpp'
+        \ : switchy#has_suffix(bufname(''), '.c') ? bufname('')[:-3] .. '.h' : bufname('')[:-5] .. '.h'
```

Comment:
Not sure if `.cpp` files are recognized of file type `C`; just in case. Same holds vice-versa for `.c` files.